Stop showing the component stack in case of an error, and show a more…

julienw opened the pull request:
… helpful message

We get too many reports with useless stacks. So now I removed the stack output and instead advise the user to look at the devtools web console. I also provided a button for an easier access to github.

I also made it localized when possible.

![image](https://user-images.githubusercontent.com/454175/225334226-10425a32-3364-46fc-a0bb-0af88fe33a81.png)

~~The localization isn't perfect, because we concatenate strings instead of providing them in their entirety. The reason is that I didn't want to spend too much time on it, and I thought that because this is an error component that the users shouldn't see much that would be good enough.~~ I think it's better now, thanks for your feedback.

julienw commented on src/components/app/ErrorBoundary.js:
```diff
@@ -61,35 +68,26 @@ export class ErrorBoundary extends React.Component<Props, State> {
           <div className="appErrorBoundaryContents">
             <div className="photon-message-bar photon-message-bar-error photon-message-bar-inner-content appErrorBoundaryMessage">
               <div className="photon-message-bar-inner-text">
                 {this.props.message}
+                {errorString ? ` (${errorString})` : null}.{' '}
+                {this.props.reportExplanationMessage}
```
@flodolo 
I know that the localization isn't perfect, because we concatenate strings instead of providing them in their entirety. The reason is that I didn't want to spend too much time on it, and I thought that because this is an error component that the users shouldn't see much that would be good enough. I hope that's good enough for you!

flodolo replied:
Can you help me understand where the `Error: ` part come from?

I'm really not a fan of having it added as part of the first sentence. Can we keep using `Details--error-boundary-message`, and insert (concatenate) a fully-formed separate sentence `Error: foo. ` instead? Or maybe put that at the very end, between the localized paragraph and the button?

julienw replied:
This comes from calling `toString` on the error object. I this case the error object was a `new Error("foo")` but usually it's an Error with more context (hopefully).

julienw replied:
Thanks for the feedback, I think this is indeed better now:

![image](https://user-images.githubusercontent.com/454175/225334394-3fa853e3-ed82-4772-91a9-2c5845ed2029.png)

Telll me what you think :-)

julienw commented on src/components/app/Root.js:
```diff
@@ -28,21 +32,28 @@ export class Root extends PureComponent<RootProps> {
+              id="Root--error-boundary-message"
+              attrs={{ message: true }}
+            >
+              <LocalizedErrorBoundary message="Uh oh, some error happened in profiler.firefox.com">
```
Here I added another one so that the non localized version would be only for errors in AppLocalizationProvider.

julienw commented on src/test/components/ErrorBoundary.test.js:
```diff
@@ -97,3 +73,26 @@ describe('app/ErrorBoundary', function () {
   });
 });
+
+describe('app/LocalizedErrorBoundary', function () {
```
I just added a small test for this new component.

canova commented on src/components/app/ErrorBoundary.js:
```diff
@@ -61,35 +68,26 @@ export class ErrorBoundary extends React.Component<Props, State> {
-                type="button"
-                onClick={this._toggleErrorDetails}
-                aria-expanded={showDetails ? 'true' : 'false'}
+                href="https://github.com/firefox-devtools/profiler/issues/new?title=An%20error%20occurred%20in%20Firefox%20Profiler"
```
I think we can even populate the content of the bug to include the error message etc. But we can do that later.

julienw commented on src/components/app/ErrorBoundary.css:
```diff
@@ -13,39 +13,19 @@
   width: 100%;
-  max-width: 600px;
-  max-height: calc(100%);
+  max-width: 800px;
```
I gave some more space to the notice.

julienw commented on src/components/app/ErrorBoundary.css:
```diff
@@ -13,39 +13,19 @@
-     time for the fade out. */
-  transition: opacity 150ms, visibility 0s 150ms;
-  visibility: hidden;
+.appErrorBoundaryInnerText > p {
```
I wanted to use `p` which is semantically better, but it comes with default margin... One day we'll probably want to include some version (or our version) of normalize.css...

canova replied:
Yeah, it would be good to have a normalize.css.

julienw commented on locales/en-US/app.ftl:
```diff
@@ -222,17 +222,19 @@ Details--open-sidebar-button =
     Please report this issue to the developers, including the full
     error as displayed in the Developer Tools’ Web Console.
 
+# This is used in a call to action button, displayed inside the error box.
```
I added some more comments here as well.

julienw commented:
oops sorry @flodolo, I missed that your r+ was just given now and I resetted it by mistake :-)

flodolo reviewed (approved):
LGTM

julienw commented:
Hey @canova, I added all new changes to a separate commit :-) Please tell me what you think!

Because I changed some photon styles, I give you the direct link to the deploy preview for the photon example page: https://deploy-preview-4526--perf-html.netlify.app/photon/.

canova reviewed (approved):
Looks good to me, thanks!
